charts/sn-platform/templates/hpa/hpa-init.yaml

```diff
@@ -0,0 +1,140 @@
+  labels:
+    {{- include "pulsar.standardLabels" . | nindent 4 }}
+---
+kind: Role
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: "{{ template "pulsar.fullname" . }}-{{ .Values.autoscaling.component }}-init-role"
+  namespace: {{ .Values.namespace }}
+  labels:
+    {{- include "pulsar.standardLabels" . | nindent 4 }}
+rules:
+  - apiGroups: [""]
+    resources: ["secrets"]
+    verbs: ["*"]
+---
+kind: RoleBinding
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: "{{ template "pulsar.fullname" . }}-{{ .Values.autoscaling.component }}-init-rolebinding"
+  namespace: {{ .Values.namespace }}
+  labels:
+    {{- include "pulsar.standardLabels" . | nindent 4 }}
+roleRef:
+  kind: Role
+  name: "{{ template "pulsar.fullname" . }}-{{ .Values.autoscaling.component }}-init-role"
+  apiGroup: rbac.authorization.k8s.io
+subjects:
+  - kind: ServiceAccount
+    name: "{{ template "pulsar.fullname" . }}-{{ .Values.autoscaling.component }}-init-serviceaccount"
+    namespace: {{ .Values.namespace }}
+---
+# the script can be found here: https://github.com/prometheus-operator/prometheus-operator/blob/release-0.25/contrib/kube-prometheus/experimental/custom-metrics-api/gencerts.sh
+# updated to take custom secret name and namespace
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: "{{ template "pulsar.fullname" . }}-{{ .Values.autoscaling.component }}-init-configmap"
+  namespace: {{ .Values.namespace }}
+  labels:
+   {{- include "pulsar.standardLabels" . | nindent 4 }}
+data:
+  gencert.sh: |
+    #!/usr/bin/env bash
+
+    # Detect if we are on mac or should use GNU base64 options
+    case `uname` in
+            Darwin)
+                b64_opts='-b=0'
+                ;;
+            *)
+                b64_opts='--wrap=0'
+    esac
+
+    go get -v -u github.com/cloudflare/cfssl/cmd/...
+
+    export PURPOSE=metrics
+    openssl req -x509 -sha256 -new -nodes -days 365 -newkey rsa:2048 -keyout ${PURPOSE}-ca.key -out ${PURPOSE}-ca.crt -subj "/CN=ca"
+    echo '{"signing":{"default":{"expiry":"87600h","usages":["signing","key encipherment","'${PURPOSE}'"]}}}' > "${PURPOSE}-ca-config.json"
+
+    export SERVICE_NAME=custom-metrics-apiserver
+    export ALT_NAMES='"custom-metrics-apiserver.monitoring","custom-metrics-apiserver.monitoring.svc"'
+    echo '{"CN":"'${SERVICE_NAME}'","hosts":['${ALT_NAMES}'],"key":{"algo":"rsa","size":2048}}' | cfssl gencert -ca=metrics-ca.crt -ca-key=metrics-ca.key -config=metrics-ca-config.json - | cfssljson -bare apiserver
+
+    cat <<-EOF > cm-adapter-serving-certs.yaml
+    apiVersion: v1
+    kind: Secret
+    metadata:
+      name: ${CERT_NAME}
+      namespace: ${CERT_NAMESPACE}
+    data:
+      serving.crt: $(cat apiserver.pem | base64 ${b64_opts})
+      serving.key: $(cat apiserver-key.pem | base64 ${b64_opts})
+    EOF
+---
+# The job basically execute above script to generate a cert for kube-aggregator to communicate with metric server.
+apiVersion: batch/v1
```

Comment:
Do there need  the control of `.Values.initialize`, I saw there are some init jobs under the control of it

Reply:
good question, after some thought, this one is a bit different, it's idempotent(end result is a secret with self signed cert for custom metric server but might with different expiration date), and it can be run multiple times, unlike other job might create meta data in zk or bk which can't be execute multiple times
also users who already has the chart installed might still want to enable hpa, but likely they can't the initialize back to false to trigger that,
so I think it's fine to not use .Values.initialize to control this job